File: frontend/src/services/stockService.js
import api from './api';

const stockService = {
  getAllStocks: (page = 0, size = 20) => {
    return api.get('/stocks', { params: { page, size } });
  },

  searchStocks: (query, page = 0, size = 20) => {
    return api.get('/stocks/search', { params: { q: query, page, size } });
  },

  getStockById: (id) => {
    return api.get(`/stocks/${id}`);
  },

  getStockBySymbol: (symbol) => {
    return api.get(`/stocks/symbol/${symbol}`);
  },

  createStock: (stockData) => {
    return api.post('/stocks', stockData);
  },

  updateStock: (id, stockData) => {
    return api.put(`/stocks/${id}`, stockData);
  },

  deleteStock: (id) => {
    return api.delete(`/stocks/${id}`);
  }
};

export default stockService;

File: backend/src/main/java/com/stockanalysis/service/StockService.java
package com.stockanalysis.service;

import com.stockanalysis.dto.StockDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface StockService {
    StockDTO createStock(StockDTO stockDTO);
    StockDTO updateStock(Long id, StockDTO stockDTO);
    StockDTO getStockById(Long id);
    StockDTO getStockBySymbol(String symbol);
    void deleteStock(Long id);
    Page<StockDTO> getAllStocks(Pageable pageable);
    Page<StockDTO> searchStocks(String searchTerm, Pageable pageable);
}
